File: app/src/main/java/com/tsp/learn/android/bestsellerbooks/BookApplication.java
package com.tsp.learn.android.bestsellerbooks;

import com.tsp.learn.android.bestsellerbooks.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

public class BookApplication extends DaggerApplication {
    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }
}

File: app/src/main/java/com/tsp/learn/android/bestsellerbooks/BookApplication.kt
package com.tsp.learn.android.bestsellerbooks

import android.app.Application
import com.tsp.learn.android.bestsellerbooks.di.AppComponent
import com.tsp.learn.android.bestsellerbooks.di.DaggerAppComponent
import com.tsp.learn.android.bestsellerbooks.di.RestModule

class BookApplication : Application() {

    companion object {
        lateinit var appComponent: AppComponent
    }

    override fun onCreate() {
        super.onCreate()
        init()
    }

    private fun init() {
        appComponent = DaggerAppComponent.builder()
            .restModule(RestModule(this))
            .build()
    }

}